next/frontend/hooks/useOAuthParams.ts

```diff
@@ -0,0 +1,45 @@
+import { Amplify } from 'aws-amplify'
+import { cognitoUserPoolsTokenProvider } from 'aws-amplify/auth/cognito'
+import { CookieStorage, defaultStorage } from 'aws-amplify/utils'
+import { useQueryState } from 'nuqs'
+import { useCallback } from 'react'
+
+import { amplifyConfig, amplifyLibraryOptions, createAmplifyConfig } from '../utils/amplifyConfig'
+import {
+  clientIdQueryParam,
+  payloadQueryParam,
+  redirectUriQueryParam,
+  stateQueryParam,
+} from '../utils/queryParamRedirect'
+
+export const useOAuthParams = () => {
+  const [clientId] = useQueryState(clientIdQueryParam)
+  const [payload] = useQueryState(payloadQueryParam)
+  const [redirectUri] = useQueryState(redirectUriQueryParam)
+  const [state] = useQueryState(stateQueryParam)
+
+  const isOAuthLogin = !!clientId
+
+  const amplifyConfigure = useCallback(() => {
```

Comment:
Function is returning `userPoolClientId`. Name of the function doesn't imply that. Change name or create new function that returns `userPoolClientId`